src/MusicalScore/Graphical/SkyBottomLineCalculator.ts

```diff
@@ -152,6 +164,43 @@ export class SkyBottomLineCalculator {
         this.mBottomLine = this.mBottomLine.map(v => (v - Math.min(...this.mBottomLine)) / unitInPixels + this.StaffLineParent.BottomLineOffset);
     }
 
+    /**
+     * go backwards through the skyline array and find a number so that
+     * we can properly calculate the average
+     * @param start
+     * @param backend
+     * @param color
+     */
+    private findPreviousValidNumber(start: number, tSkyLine: number[]): number {
+        for (let idx: number = start; idx >= 0; idx--) {
+            if (!isNaN(tSkyLine[idx]) && tSkyLine[idx] !== undefined) {
```

Comment:
we don't need the `&& tSkyLine[idx] !== undefined`, sorry, that was my mistake. isNaN() covers undefined.

Reply:
all good!  I appreciate your review!